docs/metrics.md

```diff
@@ -192,6 +195,133 @@ This module mostly contains code to load explanatory models, generate activation
+
+In order for your metric to get loaded, it must be retrievable via the following function:
+```
+metrics.get(name)
+```
+where `name` is the `str` name of your function.
+There are two ways of doing this.
+
+1) You can provide the full `.`-separated path to the metric function in the config, e.g., `"my_project.metrics.hot_dog_ness"`
+In this case, `metrics.get("my_project.metrics.hot_dog_ness")` will try to import `hot_dog_ness` from `my_project.metrics`.
+This case will only work as intended if `hot_dog_ness` is a batchwise function that outputs a list (or array) or results, one per element in the batch.
+
+2) An alternative is to use one of the existing decorators in `task` or `perturbation` to register your metric.
+These decorators, their associated namespaces, and the intended APIs of the metric functions they decorate, are:
+- `metrics.perturbation.elementwise` - `metrics.perturbation.element` - takes a single pair of `x_i` and `x_adv_i` and returns a single perturbation distance for that element.
+- `metrics.perturbation.batchwise` - `metrics.perturbation.batch` - takes a batch of `x` and `x_adv` and returns a list of results, one per data element.
+- `metrics.task.elementwise` - `metrics.task.element` - takes a single pair of `y_i` and `y_pred_i` and returns a single result for that element.
+- `metrics.task.batchwise` - `metrics.task.batch` - takes a batch of `y` and `y_pred` and returns a list of results, one per data element.
+- `metrics.task.populationwise` - `metrics.task.population` - takes lists of `y` and `y_pred` across the entire dataset and computes a single set of metrics, such as mAP.
+- `metrics.task.aggregator` - `metrics.task.aggregate` - takes a list of results from a batchwise metric and performs a non-trivial aggregation, such as for calculating total word error rate.
+
+For instance, if you were adding an accuracy metric for task results, you could do:
+```
+from armory import metrics
+@metrics.task.elementwise
+def my_accuracy_metric(y_i, y_pred_i):
+    return y_i == np.argmax(y_pred_i)
+``` 
+NOTE: when using decorators, this uses the local name of the metric, and so in this case `"my_accuracy_metric"` should be in the config, NOT `"my_project.metrics.my_accuracy_metric"`.
+It will generate an error if the name is already used.
+A different name can be used by calling the decorator method directly:
+```
+from armory import metrics
+def my_accuracy_metric(y_i, y_pred_i):
+    return y_i == np.argmax(y_pred_i)
+
+metrics.task.elementwise(my_accuracy_metric, "a_different_name")
+```
+
+Armory performs all built-in metric operations as batches, not as individual elements, so using the `elementwise` decorators will also produce a batchwise version of it that loops through the individual elements and provides a batchwise result.
+NOTE: when armory uses `get`, it will 
```

Comment:
is there text missing here? `it will...`

Reply:
Fixed.